### ISSUE TITLE
Default apphost should take platformtarget of app into account

### ISSUE THREAD

**Issue by nguerrera** [dotnet/sdk#2632]
With x64 SDK on PATH:

1. `dotnet new console`
2. Edit csproj, set PlatformTarget to x86
3. `dotnet run` (or execute apphost exe directory)_

```
Unhandled Exception: System.BadImageFormatException: Could not load file or assembly 'D:\Temp\apphost\bin\Debug\netcoreapp3.0\apphost.dll'. An attempt was made to load a program with an incorrect format
```

**Comment by nguerrera** [comment]
cc @peterhuene

**Comment by peterhuene** [comment]
Should be a simple fix to coerce a mismatch between the SDK RID and the `TargetPlatform` to what the `TargetPlatform` expects it to be.

**Comment by dasMulli** [comment]
Related to https://github.com/dotnet/sdk/issues/1905

The important question is: what is the guidance wrt which property to set?
Is `PlatformTarget` considered for default RID selection?
I'd prefer to set `RuntimeIdentifier` because it is well documented and a known "knob" for .NET Core applications. Also, NuGet should resolve correct native dependencies if needed.

**Comment by peterhuene** [comment]
I think that `RuntimeIdentifier` is the "knob" for .NET Core.  The problem here is that we have an implicit "runtime identifier" just for the apphost which only occurs when `RuntimeIdentifier` is _not_ specified.  This would only help hint at the default architecture for the apphost.

Ideally, this is going to be a temporary measure for 3.0 anyway.  I don't like having to restore for an additional runtime for "runtime agnostic" builds just to get the platform-specific apphost.  This will be how it works for the first preview of 3.0, but I'm hoping to replace this before 3.0 releases with a better mechanism that doesn't pollute your assets graph just to enable this.

**Comment by ericstj** [comment]
@peterhuene any update on this?

**Comment by livarcocc** [comment]
Not yet. This is behind the download only package work on NuGet.

**Comment by ericstj** [comment]
Does it need to be?  We could directly reference the RID-specific host packages to make sure they get downloaded for the RID-less restore, so long as the SDK can handle seeing these as runtime target items.

**Comment by nguerrera** [comment]
I think we can fix this without blocking on download only package as Peter described here: https://github.com/dotnet/sdk/issues/2632#issuecomment-434429197

**Comment by nguerrera** [comment]
I would suggest something even stronger, the fix here is separate from download only package. To pick a package, we must pick a rid. If we pick a rid wrong, it doesn't matter how we acquire the package. I would expect there to still be a concept of default rid for this feature and I would expect the fix to carry over.

**Comment by peterhuene** [comment]
I agree that I think we can fix this without relying on download-only packages.  I'll work on the fix now that I'm back from vacation.
